src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        StringBuilder report = new StringBuilder("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+        for (String name : names) {
+            int totalEarnings = 0;
+            for (String record : data) {
+                String[] parts = record.split(" ");
+                String employeeName = parts[1];
+                if (employeeName.equals(name)) {
+                    if (isDateInRange(parts[0], dateFrom, dateTo)) {
+                        totalEarnings += Integer.parseInt(parts[2]) * Integer.parseInt(parts[3]);
+                    }
+                }
+            }
+            report.append(name).append(" - ").append(totalEarnings);
+            if (!names[names.length - 1].equals(name)) {
+                report.append(System.lineSeparator());
+            }
+        }
+        return report.toString();
+    }
+
+    private static boolean isDateInRange(String dateStr, String dateFrom, String dateTo) {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy");
+        try {
+            Date date1 = sdf.parse(dateStr);
```

Comment:
use the `LocalDate` instead of `Date`

Reply:
I haven't tried LocalDate yet.

Reply:
not fixed, using LocalDate is a task requirement, you can use any guide to help you out, like this one: https://www.baeldung.com/java-8-date-time-intro
To create or parse a LocalDate, consider using this static factory method: LocalDate.parse(dateFrom, DATE_FORMATTER)